### PR TITLE
fix(platform): fdp multi combobox fires multiple times after value is selected

### LINKED ISSUE #10457: [P3.1][Sourcing] `fdp-multi-combobox` fires multiple times after value is selected
#### Is this a bug, enhancement, or feature request?

bug

#### Briefly describe your proposal.

After value is selected/deselected in `fdp-multi-combobox`, `selectionChange` emits multiple times:

https://github.com/SAP/fundamental-ngx/assets/22842279/63b95f53-d899-4142-873c-e14fa0458433

As observed from the log:

- After value is selected, it emits 4 times.
- After value is deselected, it emits 2 times.

Expectation is that it should emits only one time.

#### Which versions of Angular and Fundamental Library for Angular are affected? (If this is a feature request, use current version.)

Angular 15
0.43.8

#### If this is a bug, please provide steps for reproducing it.

- Have `fdp-multi-combobox` with selectionChange.
- Select a value and observe the log.
- Deselect a value and observe the log.

#### Please provide relevant source code if applicable.

https://stackblitz.com/edit/angular-9jp2kz-7cgxzk?file=src%2Fapp%2Fmulti-combobox-loading-example.component.ts

#### Is there anything else we should know?
It can be low/medium priority since it is not a blocker but on our application side we have to handle it carefully especially with api call.

### PR DESCRIPTION
**Pull request by skaquib123**
… selected

## Related Issue(s)

closes
#10457 
## Description
After value is selected/deselected in fdp-multi-combobox, selectionChange was emitting multiple times but now emitting only once.

## Screenshots
https://github.com/SAP/fundamental-ngx/assets/110972289/af75408a-b9c6-4210-9ac4-05440e37fa5b

##### During Implementation

1. Visual Testing:

-   [x] visual misalignments/updates
-   [x] check Light/Dark/HCB/HCW themes
-   [x] RTL/LTR - proper rendering and labeling
-   [x] responsiveness(resize)
-   [x] Content Density (Cozy/Compact/(Condensed))
-   [x] States - hover/disabled/focused/active/on click/selected/selected hover/press state
-   [x] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
-   [x] Mouse vs. Keyboard support
-   [x] Text Truncation

2. API and functional correctness

-   [x] check for console logs (warnings, errors)
-   [x] API boundary values
-   [x] different combinations of components - free style
-   [x] change the API values during testing

3. Documentation and Example validations

-   [x] missing API documentation or it is not understandable
-   [x] poor examples
-   [x] Stackblitz works for all examples

4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

##### PR Quality

-   [x] the commit message(s) follows the guideline:
        https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
-   [x] tests for the changes that have been done
-   [x] all items on the PR Review Checklist are addressed :
        https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
-   [x] Run npm run build-pack-library and test in external application
-   [x] update `README.md`
-   [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)

### CODE REVIEW

**Comment by mikerodonnell89** [comment]
PR is looking good to me, unclear as to why CI failed just now, rerunning

**Comment by mikerodonnell89** [comment]
also I assume this will need downporting

File: libs/platform/src/lib/form/multi-combobox/multi-combobox/multi-combobox.component.ts [hunk 1]
```diff
@@ -368,8 +368,6 @@ export class MultiComboboxComponent extends BaseMultiCombobox implements OnInit,
         // selected items should be displayed in the same order as options
         const valueIndexes = new Map<any, number>(this._suggestions.map((s, i) => [s.value, i]));
         this._selectedSuggestions.sort((a, b) => valueIndexes.get(a.value)! - valueIndexes.get(b.value)!);
-        this._propagateChange();
```
Comment (fkolar) [inline libs/platform/src/lib/form/multi-combobox/multi-combobox/multi-combobox.component.ts:371]: I dont think you want to remove it from here. You need to refactor it a bit.

 - When you click on the checkbox (fine you remove it as we also use `ngModelChange` to propagate the event)
 - If you dont have checkboxes and you click just on the item, then on `onOptionClicked` and in this case you removed the EventEmitter but here you want it

File: libs/platform/src/lib/form/multi-combobox/multi-combobox/multi-combobox.component.ts [hunk 2]
```diff
@@ -402,8 +400,6 @@ export class MultiComboboxComponent extends BaseMultiCombobox implements OnInit,
 
         // setting value, it will call setValue()
         this.value = selectedItems;
-
-        this._emitChangeEvent();
```
Comment (fkolar) [inline libs/platform/src/lib/form/multi-combobox/multi-combobox/multi-combobox.component.ts:406]: EventEmitter and Form's API  are two different thing, why are you removing it from here?

**Review by fkolar — CHANGES_REQUESTED** [review]

File: libs/platform/src/lib/form/multi-combobox/multi-combobox/multi-combobox.component.ts (hunk 1 above)
Comment (skaquib123) [reply libs/platform/src/lib/form/multi-combobox/multi-combobox/multi-combobox.component.ts:371]: @fkolar 
On the second point, I am not sure I understood. We don't have any component in multi-combobox that doesn't have check box except one but that also emitting values as I have checked.

**Comment by droshev** [comment]
@skaquib123 it seems that there isn't much progress here. I am closing the PR. Once you resume the work, feel free to reopen it

**Comment by skaquib123** [comment]
@droshev I was waiting for @fkolar reply as I didn't understood his point and since he didn't reply and @mikerodonnell89 commented that PR is good. Can we please merge this?

**Comment by mikerodonnell89** [comment]
@fkolar testing this locally the events are emitted once as you'd expect. But I agree that some unit tests should come with this PR

**Comment by droshev** [comment]
> @droshev I was waiting for @fkolar reply as I didn't understood his point and since he didn't reply and @mikerodonnell89 commented that PR is good. Can we please merge this?

@skaquib123 There are comments from @fkolar and there wasn't any activity or change after. I assume if Frank writes you comments, then you act and update the PR. We will close the PR soon if it stays idle

File: libs/platform/src/lib/form/multi-combobox/multi-combobox/multi-combobox.component.ts (hunk 2 above)
Comment (fkolar) [reply libs/platform/src/lib/form/multi-combobox/multi-combobox/multi-combobox.component.ts:406]: Clarified internally. 

`this.value = .. , `, calls `writeValue` and this FormAPI method already emits `selectionChange` event. so its safe to remove

**Review by fkolar — APPROVED** [review]

**Pull request merged**